src/main/java/ui/UI.java

```diff
@@ -84,7 +86,7 @@
     private TickingTimer refreshTimer;
     public GUIController guiController;
     private NotificationController notificationController;
-
+    private CountDownLatch startLatch = new CountDownLatch(1);
```

Comment:
Style-wise, I would try to make the latch use more implicit. It could probably be created in BrowserComponent and made private. UI would use it to signal by calling a BrowserComponent method.